x-pack/plugins/files/server/routes/find.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { schema, TypeOf } from '@kbn/config-schema';
+import type { Ensure } from '@kbn/utility-types';
+import type { FilesRouter } from './types';
+
+import { FindFilesHttpEndpoint, FILES_API_ROUTES } from '../../common/api_routes';
+import type { FilesRequestHandler } from './types';
+
+export const method = 'post' as const;
+
+const stringOrArrayOfStrings = schema.oneOf([schema.string(), schema.arrayOf(schema.string())]);
+
+const bodySchema = schema.object({
+  kind: schema.maybe(stringOrArrayOfStrings),
+  name: schema.maybe(stringOrArrayOfStrings),
+  mimeType: schema.maybe(stringOrArrayOfStrings),
+  status: schema.maybe(stringOrArrayOfStrings),
+  kuery: schema.maybe(stringOrArrayOfStrings),
```

Comment:
We could also always require it to be an array of strings, to simplify validation.